pkg/ebpf/encoding/format.go

```diff
@@ -8,47 +8,58 @@ import (
 )
 
 // FormatConnection converts a ConnectionStats into an model.Connection
-func FormatConnection(conn ebpf.ConnectionStats) *model.Connection {
+func FormatConnection(conn ebpf.ConnectionStats, useAddrByteString bool) *model.Connection {
 	return &model.Connection{
-		Pid:                int32(conn.Pid),
-		Laddr:              formatAddr(conn.Source, conn.SPort),
-		Raddr:              formatAddr(conn.Dest, conn.DPort),
-		Family:             formatFamily(conn.Family),
-		Type:               formatType(conn.Type),
-		TotalBytesSent:     conn.MonotonicSentBytes,
-		TotalBytesReceived: conn.MonotonicRecvBytes,
-		TotalRetransmits:   conn.MonotonicRetransmits,
-		LastBytesSent:      conn.LastSentBytes,
-		LastBytesReceived:  conn.LastRecvBytes,
-		LastRetransmits:    conn.LastRetransmits,
-		Rtt:                conn.RTT,
-		RttVar:             conn.RTTVar,
-		Direction:          formatDirection(conn.Direction),
-		NetNS:              conn.NetNS,
-		IpTranslation:      formatIPTranslation(conn.IPTranslation),
+		Pid:               int32(conn.Pid),
+		Laddr:             formatAddr(conn.Source, conn.SPort, useAddrByteString),
+		Raddr:             formatAddr(conn.Dest, conn.DPort, useAddrByteString),
+		Family:            formatFamily(conn.Family),
+		Type:              formatType(conn.Type),
+		LastBytesSent:     conn.LastSentBytes,
+		LastBytesReceived: conn.LastRecvBytes,
+		LastRetransmits:   conn.LastRetransmits,
+		Rtt:               conn.RTT,
+		RttVar:            conn.RTTVar,
+		Direction:         formatDirection(conn.Direction),
+		NetNS:             conn.NetNS,
+		IpTranslation:     formatIPTranslation(conn.IPTranslation, useAddrByteString),
```

Comment:
Have you considered custom `JSONPB(Un)Marshaler` implementations for Iptranslation and Addr in `agent-payload`  instead of passing through `useAddrByteString`?  this approach means when we parse byte-encoded messages and print them to json for debugging purposes we'll see the byte encoded IPs.  Not a huge deal since we can translate them back.